DetectDynamicJS.py

```diff
@@ -81,14 +81,17 @@ def doPassiveScan(self, baseRequestResponse):
             requestHeaders = request.tostring()[:requestBodyOffset].split('\r\n')
             requestBody = request.tostring()[requestBodyOffset:]
             modified_headers = "\n".join(header for header in requestHeaders if "Cookie" not in header)
-            newResponse = self._callbacks.makeHttpRequest(baseRequestResponse.getHttpService(), self._helpers.stringToBytes(modified_headers+requestBody))
+            newResponse = self._callbacks.makeHttpRequest(baseRequestResponse.getHttpService(), self._helpers.stringToBytes(modified_headers + requestBody))
+            newResponseInfo = self._helpers.analyzeResponse(newResponse.getResponse())
+            if not (self.hasValidStatusCode(newResponseInfo.getStatusCode()) and self.hasBody(newResponseInfo.getHeaders())):
```

Comment:
There are also many frameworks that set a cookie upon initial contact. If then no cookie is present, no static files are served.
But the potential for a login oracle should be reported. As it is not an actual dynamic java script I would add a new issue.